sdk/tests/network-client.test.ts

```diff
@@ -198,6 +198,27 @@ describe('NodeConnection', () => {
         });
     });
 
+    describe('waitForTransactionConfirmation', () => {
+        it('should return accepted for a valid tx ID', async () => {
+          const connection = new AleoNetworkClient("https://api.explorer.provable.com/");
+          const status = await connection.waitForTransactionConfirmation("at1dl9lze8wscct0dee8x9tjnfmpjvj33hh23jcnp5f0ywjn5552yrsperzl9");
+          expect(status).to.equal("accepted");
+        });
+
+        it('should return rejected for a rejected tx ID', async () => {
+          const connection = new AleoNetworkClient("https://api.explorer.provable.com/");
+          const status = await connection.waitForTransactionConfirmation("at1x5ed0pyjpt3770e0m60psvdvqeww5z5f32t7velrmrjfhvzgysxqll4g6a");
+          expect(status).to.equal("rejected");
+        });
+      
+        it('should throw for an invalid tx ID', async () => {
+          const connection = new AleoNetworkClient("https://api.explorer.provable.com/");
+          await expectThrows(() =>
+            connection.waitForTransactionConfirmation("at1dl9lze8wscct0dee8x9tjnfmpj12345678jcnp5f0ywjn5552yrsperzl9")
```

Comment:
This is a good test, but we should ensure, that the main method detects a malformed transaction ID.

In the case of a malformed URL, you'll get responses like:
`Invalid URL: Invalid byte size for a bech32m hash: 6 bytes`
`Invalid URL: invalid checksum`
`Invalid URL: invalid character (code=b)`

So the main method should detect an invalid URL by the presence of `Invalid URL` in the response. And communicate that back to the user in the error.